### PR TITLE
Various network fixes

### LINKED ISSUE #3061: Allow user to configure link properties
The following properties are currently hardcoded:

https://github.com/oxidecomputer/omicron/blob/1e89e384a2e1186bd6bf33ebfce4f0d8a5d24ce0/nexus/src/app/sagas/switch_port_settings_apply.rs#L147-L154

### LINKED ISSUE #4458: Link config API does not allow for setting link autonegotiation
The following should include an `autonegotiation` parameter.

https://github.com/oxidecomputer/omicron/blob/6db881937cc4c303c81a6e1ba8bb8c9210a90dba/nexus/types/src/external_api/params.rs#L1364-L1378

### LINKED ISSUE #4470: Link parameters are not making it from RSS to the nexus database
In the code below, we're not adding the link parameters to the Nexus DB once the rack has been initialized.

https://github.com/oxidecomputer/omicron/blob/03c7f12cf7a70ed7b360a310e7a6b46d0075ac72/nexus/src/app/rack.rs#L459-L468

### PR DESCRIPTION
**Pull request by rcgoodfellow**
- Fixes #4470
- Fixes #4458
- Fixes #3061

### CODE REVIEW

**Comment by rcgoodfellow** [comment]
Using the same RSS configuration:

Here is what we see from the API right after a rack is commissioned.

```
oxide system networking switch-port-settings viewport default-uplink0
success
SwitchPortSettingsView {
    addresses: [
        SwitchPortAddressConfig {
            address: V4(
                Ipv4Net(
                    "198.51.101.2/30",
                ),
            ),
            address_lot_block_id: ccf518cf-fae6-4687-97de-6654e2e0d010,
            interface_name: "phy0",
            port_settings_id: 70905d7e-7425-4fc8-b97b-e3b0ca29985e,
        },
    ],
    bgp_peers: [],
    groups: [],
    interfaces: [],
    link_lldp: [],
    links: [],
    port: SwitchPortConfig {
        geometry: Qsfp28x1,
    },
    routes: [],
    settings: SwitchPortSettings {
        description: "initial uplink configuration",
        id: 70905d7e-7425-4fc8-b97b-e3b0ca29985e,
        name: Name(
            "default-uplink0",
        ),
        time_created: 2023-11-28T03:26:52.109575Z,
        time_modified: 2023-11-28T03:26:52.109575Z,
    },
    vlan_interfaces: [],
}
```

Here is what it looks like with this PR. Note that the BGP peer info and link info are now present.

```
$ oxide system networking switch-port-settings viewport default-uplink0
ry@rymb:~$ oxide system networking switch-port-settings viewport default-uplink0
success
SwitchPortSettingsView {
    addresses: [
        SwitchPortAddressConfig {
            address: V4(
                Ipv4Net(
                    "198.51.101.2/30",
                ),
            ),
            address_lot_block_id: c8aa667b-7835-4dd7-907d-21706c758d9e,
            interface_name: "phy0",
            port_settings_id: 91df9f00-d35d-45eb-83cf-9d161432eff8,
        },
    ],
    bgp_peers: [
        SwitchPortBgpPeerConfig {
            addr: 198.51.101.1,
            bgp_config_id: d3bcc92a-adf3-4155-a44d-c5894daf3b3d,
            interface_name: "phy0",
            port_settings_id: 91df9f00-d35d-45eb-83cf-9d161432eff8,
        },
    ],
    groups: [],
    interfaces: [],
    link_lldp: [
        LldpServiceConfig {
            enabled: false,
            lldp_config: None,
        },
    ],
    links: [
        SwitchPortLinkConfig {
            link_name: "phy",
            lldp_service_config_id: e14453f9-6293-4cd3-96d5-b2564977d1bb,
            mtu: 1500,
            port_settings_id: 91df9f00-d35d-45eb-83cf-9d161432eff8,
        },
    ],
    port: SwitchPortConfig {
        geometry: Qsfp28x1,
    },
    routes: [],
    settings: SwitchPortSettings {
        description: "initial uplink configuration",
        id: 91df9f00-d35d-45eb-83cf-9d161432eff8,
        name: Name(
            "default-uplink0",
        ),
        time_created: 2023-11-28T22:27:40.087932Z,
        time_modified: 2023-11-28T22:27:40.087932Z,
    },
    vlan_interfaces: [],
}
```

File: tools/generate-wicketd-api.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+./target/debug/wicketd openapi > openapi/wicketd.json
```
Comment (internet-diglett) [inline tools/generate-wicketd-api.sh:3]: Is this a convenience script, or is it called from some of our build machinery?

**Review by internet-diglett — APPROVED** [review]
LGTM!

File: tools/generate-wicketd-api.sh (hunk 1 above)
Comment (rcgoodfellow) [reply tools/generate-wicketd-api.sh:3]: Convenience script. There are others like it for nexus and sled agent.

**Pull request merged**